apps/jenkins/src/main/fabric8/openshift-deployment.yml

```diff
@@ -60,8 +60,16 @@ spec:
           value: "true"
         - name: KUBERNETES_MASTER
           value: "https://kubernetes.default:443"
-        - name: MAX_METASPACE_SIZE
-          value: 200m
+        - name: JAVA_GC_OPTS
+          value: "-XX:+UseParallelGC -XX:MinHeapFreeRatio=5 -XX:MaxHeapFreeRatio=10 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90"
+        - name: JAVA_OPTS
+          value: "-XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -Dsun.zip.disableMemoryMapping=true"
+        - name: OPENSHIFT_JENKINS_JVM_ARCH
+          value: "i686"
```

Comment:
if you're using our jenkins image you shouldn't need this, we autodetect the heap size and pick the arch appropriately.

Reply:
We're using a fork https://github.com/fabric8io/jenkins-openshift-base, I think initially so we could release it ourselves and have a fixed tag version.  We need to rebase as we're way out of date and also PR any changes that we've made which you'd want.  i.e. you probably don't want the stackdriver agent and certain other changes but we can start with a PR and see what you'd like.

Going forward we can set up a pipeline job to track the upstream openshift repo to make sure we keep up to date.

Reply:
Yes, the `OPENSHIFT_JENKINS_JVM_ARCH` setting is there only to be explicit. As to the `UseCGroupMemoryLimitForHeap` flag, it's the same deal, really. The point is that those settings should work reasonably well for other JVM containers without the extra magic you do in the scripts.